easybuild/easyconfigs/c/cwltool/cwltool-3.1.20221018083734-foss-2021a.eb

```diff
@@ -0,0 +1,108 @@
+##
+# This is a contribution from SIB Swiss Institute of Bioinformatics
+# Homepage:     https://www.sib.swiss/research-infrastructure/competence-centers/vital-it
+#
+# Authors::     Sebastien Moretti <sebastien.moretti@sib.swiss>
+#
+##
+
+easyblock = 'PythonBundle'
+
+name = 'cwltool'
+version = '3.1.20221018083734'
+
+homepage = 'https://github.com/common-workflow-language/cwltool'
+description = """Common workflow language (CWL) reference implementation."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+builddependencies = [
+    ('binutils', '2.36.1'),
+    ('CMake',    '3.20.1'),
+    ('make',     '4.3'),
+    ('Check',    '0.15.2'),
+    ('SWIG',     '4.0.2'),
+    ('expat',    '2.2.9'),
+    ('bzip2',    '1.0.8'),
+    ('zlib',     '1.2.11'),
+]
+
+# The Qt5 dependency of Graphviz (required by pydot) may be long to build and
+# requires a lot of disk space and memory!
+dependencies = [
+    ('Python',            '3.9.5'),
+    ('typing-extensions', '3.10.0.0'),
+    ('pydot',             '1.4.2'),
+    ('lxml',              '4.6.3'),
+    ('ruamel.yaml',       '0.17.21'),
+    ('networkx',          '2.5.1'),
+]
+
+use_pip = True
+
+# Fetch the tar.gz, not the whl files!
+exts_list = [
+    ('argcomplete', '2.0.0', {
+        'checksums': ['6372ad78c89d662035101418ae253668445b391755cfe94ea52f1b9d22425b20'],
+    }),
+    ('humanfriendly', '10.0', {
+        'checksums': ['6b0b831ce8f15f7300721aa49829fc4e83921a9a301cc7f606be6686a2288ddc'],
+    }),
+    ('coloredlogs', '15.0.1', {
+        'checksums': ['7c991aa71a4577af2f82600d8f8f3a89f936baeaf9b50a9c197da014e5bf16b0'],
+    }),
+    ('isodate', '0.6.1', {
+        'checksums': ['48c5881de7e8b0a0d648cb024c8062dc84e7b840ed81e864c7614fd3c127bde9'],
+    }),
+    ('rdflib', '6.2.0', {
+        'checksums': ['62dc3c86d1712db0f55785baf8047f63731fa59b2682be03219cb89262065942'],
+    }),
+    ('mistune', '0.8.4', {
+        'checksums': ['59a3429db53c50b5c6bcc8a07f8848cb00d7dc8bdb431a4ab41920d201d4756e'],
+    }),
+    ('schema-salad', '8.3.20221016151607', {
```

Comment:
Thanks for this!

Please consider either using the pre-built binary wheels for schema-salad, or compile your own using `mypyc`

Examples from other packaging systems

https://github.com/conda-forge/schema-salad-feedstock/blob/b4719fc95e2e4f8ebf85e2c266af0983312ddc3a/recipe/meta.yaml#L17
https://github.com/common-workflow-language/schema_salad/blob/b86efc78a4dc4c1f4055ab24615504e177e7e053/schema_salad.Dockerfile#L9

Reply:
@mr-c Can you elaborate on the benefit of using `mypyc` instead of a regular `pip install` for `schema-salad`?

Reply:
mypyc can be used to compile Python projects with type hints into fast binary modules. For example, the Python code formatter `black` is distributed in a binary wheel this way (despite being a pure Python codebase)

https://pypi.org/project/black/#files

https://mypyc.readthedocs.io/en/latest/introduction.html